src/order/atoms.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author:  Aaron Anderson.
+-/
+
+import order.bounded_lattice
+import order.order_dual
+
+/-!
+# Atoms, Coatoms, and Simple Lattices
+
+This module defines atoms, which are minimal non-`⊥` elements in bounded lattices, simple lattices,
+which are lattices with only two elements, and related ideas.
+
+## Main definitions
+  * `is_atom a` indicates that the only element below `a` is `⊥`.
+  * `is_coatom a` indicates that the only element above `a` is `⊤`.
+  * `is_simple_lattice` indicates that a bounded lattice has only two elements, `⊥` and `⊤`.
+
+-/
+
+variable {α : Type*}
+
+section atoms
+
+section is_atom
+
+variable [order_bot α]
+
+/-- An atom of an `order_bot` is an element with no other element between it and `⊥`,
+  which is not `⊥`. -/
+def is_atom (a : α) : Prop := a ≠ ⊥ ∧ (∀ b, b < a → b = ⊥)
+
+lemma eq_bot_or_eq_of_le_atom {a b : α} (ha : is_atom a) (hab : b ≤ a) : b = ⊥ ∨ b = a :=
+or.imp_left (ha.2 b) (lt_or_eq_of_le hab)
+
+end is_atom
+
+section is_coatom
+
+variable [order_top α]
+
+/-- A coatom of an `order_top` is an element with no other element between it and `⊤`,
+  which is not `⊤`. -/
+def is_coatom (a : α) : Prop := a ≠ ⊤ ∧ (∀ b, a < b → b = ⊤)
+
+lemma eq_top_or_eq_of_coatom_le {a b : α} (ha : is_coatom a) (hab : a ≤ b) : b = ⊤ ∨ b = a :=
+or.imp (ha.2 b) eq_comm.2 (lt_or_eq_of_le hab)
+
+end is_coatom
+
+variables [bounded_lattice α] {a : α}
+
+lemma is_atom_iff_is_coatom_dual : is_atom a ↔ is_coatom (order_dual.to_dual a) := iff.refl _
+
+lemma is_coatom_iff_is_atom_dual : is_coatom a ↔ is_atom (order_dual.to_dual a) := iff.refl _
+
+end atoms
+
+/-- A lattice is simple iff it has only two elements, `⊥` and `⊤`. -/
+class is_simple_lattice (α : Type*) [bounded_lattice α] extends nontrivial α : Prop :=
+(eq_bot_or_eq_top : ∀ (a : α), a = ⊥ ∨ a = ⊤)
+
+export is_simple_lattice (eq_bot_or_eq_top)
+
+theorem is_simple_lattice_iff_is_simple_lattice_order_dual [bounded_lattice α] :
+  is_simple_lattice α ↔ is_simple_lattice (order_dual α) :=
+begin
+  split; intro i; haveI := i,
+  { exact { exists_pair_ne := @exists_pair_ne α _,
+      eq_bot_or_eq_top := λ a, or.symm (@eq_bot_or_eq_top α _ _ a) } },
```

Comment:
Why do you need the `@` here? Is the `export` doing something stupid with the binders? If so, please restate it by hand.

Reply:
The `export` doesn't really seem to be changing anything.
I was able to express this neatly with `order_dual.to_dual`, but I still had to do a bit of a weird trick for `order_dual.of_dual`. The same was true with the lemma exported manually.